### DISCUSSION TITLE
Week 7 Report

### DISCUSSION THREAD

**Discussion by derrrobin** [SE-TINF22B6/CookPit#34] [General]
`In this weekly report we focus on the explanation of our class diagram and our working progress in this week.`

# Cookpit Class Diagram with Builder Pattern

## Overview

We've designed a class diagram for our cooking website 'Cookpit' utilizing the powerful Builder Pattern. This architectural choice is driven by the necessity to construct intricate objects like cakes and drinks, where attributes may vary based on different types.

## Class Hierarchy

- **Item**
  - *Cake*
    - *VeganCake*
    - *GlutenFreeCake*
  - *Drink*
    - *ColdDrink*
    - *WarmDrink*

## Builder Pattern in Action

The Builder Pattern empowers us to manage the step-by-step creation of these complex objects. For example, the construction process for a vegan cake differs from that of a gluten-free cake. This design ensures a clear and flexible interface for crafting diverse cakes and drinks.

## Specialized Classes

To streamline the construction process, we've introduced specialized classes with a own direction:
- **Cake direction**
- **Drink direction**

These directors coordinate with concrete builders, guaranteeing the correct construction of objects.

## Result

The outcome is a clear and maintainable structure that facilitates the representation and management of various dishes and drinks on our innovative cooking website, 'Cookpit.'

You can find our diagram here: https://github.com/SE-TINF22B6/CookPit/blob/main/Docs/SRS%20Images/ClassDiagram_designPatterns.png

# Our progress this week

To allow our users, to log intop their accounts and for example to save recipies, we created an first Login popup with all essential features.
<br>
![image](https://github.com/SE-TINF22B6/CookPit/assets/122603912/154634c7-0e35-4e09-94f0-e881a35a1663)

**Comment by dolerich-hirnfiedler** [comment]
Dear Cookpit Team
 I am thrilled to see, that you already use advanced object-oriented concepts like the builder pattern. However, I can't see how this helps to enforce the SOLID principle based on the block post.  I am very curious, what the benefits of the builder pattern are for your usecase.

**Reply by derrrobin** [reply]
Hey, thank you for your feedback. The problem that we had was that none of the design elements fit because we are not object orientated programmers and so we had a problem to find a special design element. Nevertheless we found that this method fits best and so we decided to use it. Thanks for the feedback though, we will try to pay attention to this in the future.

**Comment by LaggyNacho** [comment]
Dear Cookpit Team 
The latest update on the Cookpit project is really good. You used something called the Builder Pattern to make their website better at handling different kinds of foods like cakes and drinks. This may make the website easier to manage and use. Also, you added a new login feature, which is a nice touch for people using the site. I think you did a great job in making the site more user-friendly and organized. I also find the choice of design principle interesting, maybe you could give more detailed information about it.

**Comment by MaarcS** [comment]
Dear CookPit team,

Your class diagram is very extensive, and i can see that you put a lot thoughts into it. I also really like the design of your login, which fits well to a cooking site.
I see that you chose a design principle that was not mentioned in the lecture. I would have liked it if you included a more detailed exlanation of it (maybe with an example or where to find it in your diagram) so it becomes more clear for people unfamilar with the pattern.
Nevertheless, nice blog post!

Your competitors,
CookHub
